### PR TITLE
importccl: Move the desc preparation from planHook to IMPORT job.

### PR DESCRIPTION
**Pull request by adityamaru27**
This change moves the steps taken to prepare the descriptors
before starting ingestion, from the planner to the IMPORT job.
The preparation phase involves writing new descs if importing
into new tables, and taking them offline. The motivation for
this is a bigger refactor to make IMPORT job have distinct
stages which can be reflected in its status.

There are no changes to the logic except for when we
ascertain a Walltime to import. For new tables this is done
as soon as the descs are "prepared", whereas when importing
into existing tables we wait until all nodes see the same version
of the modified desc.

### CODE REVIEW

**Review by dt — APPROVED** [review]
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39078)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru27, @dt, and @lucy-zhang)

___
*[pkg/ccl/importccl/import_stmt.go, line 673 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/39078#-Lk_n59c4TL6proaWRip:-Lk_n59c4TL6proaWRiq:b-br0sod) ([raw file](https://github.com/cockroachdb/cockroach/blob/91221dc63ded6a627c7e3e85a0bd7775ac1916cd/pkg/ccl/importccl/import_stmt.go#L673)):*
> ```Go
> 	}
> 
> 	// TODO(dt): we should be creating the job with this txn too. Once a job
> ```

I think this change can delete this TODO.

**Review by adityamaru27 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/39078)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @lucy-zhang)

___
*[pkg/ccl/importccl/import_stmt.go, line 673 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/39078#-Lk_n59c4TL6proaWRip:-LkdfkqzA76XVxYjHATU:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/91221dc63ded6a627c7e3e85a0bd7775ac1916cd/pkg/ccl/importccl/import_stmt.go#L673)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

I think this change can delete this TODO.
</blockquote></details>

Done.

**Review by thoszhang — APPROVED** [review]

File: pkg/ccl/importccl/import_stmt.go [hunk 1]
```diff
@@ -712,6 +608,179 @@ type importResumer struct {
...
+		// wait for all nodes to see the same descriptor version before doing so.
+		if !hasExistingTables {
+			importDetails.Walltime = p.ExecCfg().Clock.Now().WallTime
+		} else {
+			importDetails.Walltime = 0
+		}
+
+		// Update the job once all descs have been prepared for ingestion.
+		if err = r.job.WithTxn(txn).SetDetails(ctx, importDetails); err != nil {
+			return err
```
Comment (dt) [inline pkg/ccl/importccl/import_stmt.go]: ugh, for some reason our linter doesn't like this (err check here then nil return below, and just wants you to return it)

**Comment by adityamaru27** [comment]
TFTR!
bors r+

**Pull request merged**